Project/Source/Components/UI/ComponentText.cpp

```diff
@@ -47,26 +49,49 @@ void ComponentText::Init() {
 
 void ComponentText::OnEditorUpdate() {
 	ImGuiInputTextFlags flags = ImGuiInputTextFlags_AllowTabInput;
+
+	bool mustRecalculateVertices = false;
+
 	if (ImGui::InputTextMultiline("Text input", &text, ImVec2(-FLT_MIN, ImGui::GetTextLineHeight() * 8), flags)) {
 		SetText(text);
 	}
 	ImGui::ResourceSlot<ResourceShader>("shader", &shaderID);
 	UID oldFontID = fontID;
 	ImGui::ResourceSlot<ResourceFont>("Font", &fontID);
 	if (oldFontID != fontID) {
-		RecalculcateVertices();
+		mustRecalculateVertices = true;
 	}
 
-	if (ImGui::DragFloat("Font Size", &fontSize, 0.2f, 0, FLT_MAX)) {
-		RecalculcateVertices();
+	if (ImGui::DragFloat("Font Size", &fontSize, 2.0f, 0, FLT_MAX)) {
+		mustRecalculateVertices = true;
 	}
+	if (ImGui::DragFloat("Line height", &lineHeight, 2.0f, 0, FLT_MAX)) {
+		mustRecalculateVertices = true;
+	}
+
+	int type = static_cast<int>(textAlignment);
+	ImGui::RadioButton("Left", &type, 0);
+	ImGui::SameLine();
+	ImGui::RadioButton("Center", &type, 1);
+	ImGui::SameLine();
+	ImGui::RadioButton("Right", &type, 2);
+	textAlignment = static_cast<TextAlignment>(type);
+
 	ImGui::ColorEdit4("Color##", color.ptr());
+
+	if (mustRecalculateVertices) {
+		RecalculcateVertices();
```

Comment:
BUG: Init does not make a call to RecalculateVertices() making loading a scene where this component is found not show the proper visuals until the components are selected and the OnEditorUpdate is called.

To reproduce: Save a scene with one or more texts, having set their values to something visually pleaseant, close engine, open engine, load scene.

Reply:
**Not a bug.** It does call RecalculateVertices, but the scale is not correctly calculated because screenFactor is 1 at the point the scene is loaded. Opening the screenFactor bug task https://app.clickup.com/t/j53evc.